### ISSUE TITLE
Add inline font Appearance setting to 'More' in Paragraph block

### ISSUE THREAD

**Issue by ellenbauer** [WordPress/gutenberg#54680]
Currently, it does not seem possible to highlight a section of a paragraph block with a font weight other than bold. 

In the following example, I would like to highlight the text in a semibold font weight:

<img width="467" alt="Screenshot 2023-09-21 at 1 18 41 PM" src="https://github.com/WordPress/gutenberg/assets/17613630/b3067d16-8037-4ce4-97f0-de744a8805f3">

## What is your proposed solution?
It would be great to have the option to add Appearance under the list of More actions so I can alter the text inline similar to the already available Highlight color option:

<img width="531" alt="Screenshot 2023-09-21 at 1 27 24 PM" src="https://github.com/WordPress/gutenberg/assets/17613630/6615065e-0338-42c1-b7f8-a23e2c65f971">

**Comment by jasmussen** [comment]
Coming back to this one, thanks for the issue @ellenbauer. Help me unpack the details, because I perceive two aspects to it.

One is the use case for a font weight other than "bold"—which should not be confused with the "Bold" button which simply adds the semantic `strong` tag. Do you feel these need to be changeable on an _inline_ basis, i.e. per-paragraph? Or is this rather something you'd like to see possible to style in Global Styles? I.e. blanket rules for the appearance of the bold and italic buttons, for example? Think: like how you would style links: underlined, or bold, or both.

The other is the use case for **inline formatting tools in general**. I've found myself wanting this in a few cases, such as for [the release sites](https://wordpress.org/download/releases/7-0/). I wonder here if we can do something simple:

* If your caret is resting on a paragraph, or if you've selected all text inside it, inspector options affect the whole block.
* If you have made a text selection, inspector options affect only that text selection.

<img width="3042" height="1714" alt="Image" src="https://github.com/user-attachments/assets/815c9247-3b1b-46c8-a9f2-0d03afa49bd2" />

It would be a UI-less improvement that could be kinda nice. What do you think, @WordPress/gutenberg-design ?

**Comment by RatzeR** [comment]
Hi @jasmussen, 

as https://github.com/WordPress/gutenberg/issues/57583 is closed I just want to add my 2 cents here. 

My point was, that it is not possible to control the font-weight of the strong tag in the theme.json. Currently it is only possible via custom CSS. 

By default the strong tag uses "bolder" which the browser translates by default to font-weight: 700. 
If you are using a very thin font-weight as default - for example 200 or 300 then 700 might be to much an you want to use 500, which is not possible via theme.json.

> If you have made a text selection, inspector options affect only that text selection.

This would be a nice solution, but this should also be supported in headlines and it is not solving my original problem - it should be possible to control the default font-weight as well.

**Comment by jasmussen** [comment]
Thanks for adding that. That's definitely part of it. 

How would you feel about simply adding "Bold" and "Italic" to global styles, as elements just like text and link here?

<img width="694" height="828" alt="Image" src="https://github.com/user-attachments/assets/871bb7db-19eb-42a4-ab93-fea2888ab778" />

Depending on the feedback on this issue, we might even want to revisit your now-closed issue, and reopen+rephrase it to be just about that.

**Comment by RatzeR** [comment]
That sounds like it would work well.  As far as I know, those values are also available in the theme.json, so this would be the perfect solution.

**Comment by jasmussen** [comment]
Cool, let's hear details of this main issue and what it delves into, and either refocus this issue on that, or reopen yours.
